### PR TITLE
Skip tests when we can not use runas

### PR DESCRIPTION
**Pull request by dwoz**
### What does this PR do?

Skip tests that require runas when we are running under a limited winrm shell.

### Commits signed with GPG?

Yes

### CODE REVIEW

**Review by cachedout — APPROVED** [review]

**Comment by dwoz** [comment]
re-run py

**Comment by dwoz** [comment]
re-run all

**Review by twangboy — APPROVED** [review]

**Review by gtmanfred — APPROVED** [review]

**Comment by dwoz** [comment]
re-run lint

**Comment by rallytime** [comment]
re-run lint

File: tests/integration/states/test_pip_state.py [hunk 1]
```diff
@@ -37,13 +37,34 @@
...
+    '''
+    Detect if we are running in a limited shell (winrm) and are un-able to use
+    the runas utility method.
+    '''
+    if salt.utils.is_windows():
+        try:
+            salt.utils.win_runas.run_as(
+                'cmd.exe /c echo 1', 'noexistuser', 'n0existp4ss',
+            )
+        except WindowsError as exc:  # pylint: disable=E060
```
Comment (rallytime) [inline tests/integration/states/test_pip_state.py]: Looks like the linter doesn't like `E060`. You should be able to replace that just with `undefined-variable` instead.

**Review by rallytime — COMMENTED** [review]
I linted this by hand since the lint run is down. Can you make that change really quickly? Then we can get this in.

**Pull request merged**